### DISCUSSION TITLE
VS Code problems

### DISCUSSION THREAD

**Discussion by yadanilov19** [likec4/likec4#626] [Q&A] [answered]
Hi!
I've faced with problem with language server (I guess)

I renamed folder with .c4 file and then everything in this file were broken

Here's steps to reproduce the problem:

1. Create landscape.c4 file with some definitions
2. Create a folder above (e.x. test)
3. Create some c4 file inside the test folder with some definitions
4. Rename test folder to something new (e.x. test2)

After that you will see a duplication error  like that
![image](https://github.com/likec4/likec4/assets/17161065/a6f8b2fa-6476-4b64-a4ce-a8979df5349a)

**Comment by davydkov** [comment, marked as answer]
Hey @yadanilov19 

I apologize for not getting back to you earlier. 
I saw the same issue in Langium repository.

I recall the issue was that langium did not subscribe to vscode workspace files changes, but only to TextDocument changes. And I thought they fixed in 3.x, but apparently not. Let me check on this.

Meanwhile, what you can do:
- Run command `LikeC4: Rebuild` (mostly helps 😄 )
- Close/Open vscode (sorry for such inconveniences, but this helps always)

**Comment by yubrshen** [comment]
I also found that I have to restart VS Code when creating a new design, otherwise, elements of the previous design would show up in the new design's.
It would be nice that those old elements would not show up in the new design's. How to achieve that without restarting VS Code?
